### PR TITLE
publishing workflow

### LINKED ISSUE #15214: release workflow need to allow releasing all pkgs at once
### Bug description

We have not tested releasing all packages at once seems that once a job is skipped it does not continue if others assume it as required/depends-on

### How to reproduce the bug

https://github.com/Lightning-AI/lightning/actions/runs/3289378989/jobs/5420836621

### Error messages and logs

```

# Error messages and logs here please

```

### Environment

```

#- Lightning Component (e.g. Trainer, LightningModule, LightningApp, LightningWork, LightningFlow):
#- PyTorch Lightning Version (e.g., 1.5.0):
#- Lightning App Version (e.g., 0.5.2):
#- PyTorch Version (e.g., 1.10):
#- Python version (e.g., 3.9):
#- OS (e.g., Linux):
#- CUDA/cuDNN version:
#- GPU models and configuration:
#- How you installed Lightning(`conda`, `pip`, source):
#- Running environment of LightningApp (e.g. local, cloud):

```

### More info

_No response_

### PR DESCRIPTION
**Pull request by Borda**
## What does this PR do?

Update workflow such that you can release packages
Resolves #15214

## Before submitting

- [x] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/Lightning-AI/lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [ ] Did you make sure to **update the documentation** with your changes? (if necessary)
- [ ] Did you write any **new necessary tests**? (not for typos and docs)
- [ ] Did you verify new and **existing tests pass** locally with your changes?
- [ ] Did you list all the **breaking changes** introduced by this pull request?
- [ ] Did you **update the CHANGELOG**? (not for typos, docs, test updates, or minor internal changes/refactors)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/Lightning-AI/lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

cc @carmocca @akihironitta @borda @tchaton @rohitgr7

### CODE REVIEW

File: .github/workflows/release-pypi.yml [hunk 1]
```diff
@@ -84,89 +61,9 @@ jobs:
         name: dist-packages-${{ github.sha }}
         path: dist
 
-  # based on https://github.com/pypa/gh-action-pypi-publish
-  download-package:
-    needs: releasing
+  upload-packagea:
```
Comment (otaj) [inline .github/workflows/release-pypi.yml:64]:
```suggestion
  upload-packages:
```

File: .github/workflows/release-pypi.yml [hunk 2]
```diff
@@ -74,7 +51,7 @@ jobs:
 
     - name: Build packages
       env:
-        PACKAGE_NAME: ${{ matrix.pkg }}
+        PACKAGE_NAME: ${{ matrix.pkg-name }}
       run: |
         python setup.py sdist bdist_wheel
```
Comment (otaj) [inline .github/workflows/release-pypi.yml:56]: Before we start building and uploading and all, we still need to get `setup_tools.py` from LAI branch to master

**Review by otaj — APPROVED** [review]

File: .github/workflows/release-pypi.yml (hunk 2 above)
Comment (Borda) [reply .github/workflows/release-pypi.yml:56]: Yes, this needs to be merged after #15105

**Review by carmocca — APPROVED** [review]
How are we testing that this works?

**Comment by Borda** [comment]
> How are we testing that this works?

we'll test it with RC :)

**Pull request merged**